### PR TITLE
Dependency submission CI

### PR DESCRIPTION
**Pull request by Ao-senXiong**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -159,6 +159,8 @@ jobs:
     runs-on: ubuntu-latest
     permissions:
       contents: write
+    # Only run on the main repository because forks do not have write access to the main repository.
```
Comment (wmdietl) [inline .github/workflows/ci.yml:162]: There is a lot more discussion here https://github.com/gradle/actions/blob/main/docs/dependency-submission.md#usage-with-pull-requests-from-public-forked-repositories
I'm not sure what we lose by disabling this here compared to doing the three separate jobs mentioned in that explanation. Can you take a look?

Comment (Ao-senXiong) [reply .github/workflows/ci.yml]: I updated with three separate workflow. If we just disable it, there might be security issue in the PR and get merged into our repo.

**Comment by Ao-senXiong** [comment]
I don't understand this junit test failing on JDK 21. It seems work fine on Chris's PR.

**Comment by Ao-senXiong** [comment]
Make name consistent, and add link to yml file.

**Comment by Ao-senXiong** [comment]
> Make name consistent, and add blog link `https://github.com/gradle/actions/blob/main/docs/dependency-submission.md#usage-with-pull-requests-from-public-forked-repositories` to yml file.

Resolved.

**Review by wmdietl — APPROVED** [review]
Thanks!

**Pull request merged**